### DISCUSSION TITLE
Is there a way to download MAG/UBA sequences from the Genbank?

### DISCUSSION THREAD

**Discussion by songb1230** [kblin/ncbi-genome-download#184] [Q&A]
Dear kblin,

I like to build a database of MAG/UBA sequences reported in the papers of Parks et al 2017 (https://www.nature.com/articles/s41564-017-0012-7) and Anantharaman et al 2016 (https://www.nature.com/articles/ncomms13219). Is there a way to download the fasta and genbank files of the MAG/UBA sequences?

Thanks

BK

**Comment by kblin** [comment]
I've just looked at the Parks et al. paper, but for the UBA set there, the authors have created a BioProject at NCBI (https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA348753). On that page, there's an "assembly details" table, with a download link that will give you what looks like a tab-separated format containing assembly IDs in the first column.

With a bit of scripting magic, you could turn that into an assembly ID file suitable for `ncbi-genome-download`. Something like this (I'm using bash, but anything that works for you is good):

```bash
# use tail to get rid of the two lines of comments on top
# then use cut to only keep the first column
tail -n +3 PRJNA348753_AssemblyDetails.txt | cut -f1 > assembly_ids.txt
```

Now you can use that to run `ncbi-genome-download` (make sure you have at least version 0.2.7 installed):

```bash
ncbi-genome-download --section genbank --assembly-accessions assembly_ids.txt bacteria,archaea
```

Hope this helps.

**Comment by songb1230** [comment]
Yes, it works.

Thank you

**Comment by kblin** [comment]
Let's just keep this open in case anybody else wants to do something similar.

**Comment by songb1230** [comment]
Sounds good.
Thanks

BK Song

**Comment by EBosi** [comment]
Hi Kai,
I read in the [ncbi FAQs](https://www.ncbi.nlm.nih.gov/genbank/wgsfaq/#metagen) that the Biosample entry of MAGs should have the tag sample_type set as metagenomic assembly, is it something that can be accessed with ncbi-genome-download? If yes it would be possible to use that filter to download MAGs.

**Comment by chingciripit** [comment]
Hi Kai,

Is there a way to automate the download of PRJNA348753_AssemblyDetails.txt from the NCBI Bioproject website by using only the BioProject ID PRJNA348753 as the input?

Thanks.

Ching
